### PR TITLE
Add example for using the class previously created for Postgres

### PR DESCRIPTION
**Pull request by AkshathaLaxmi**
SAM is really really shitty. Had to do a lot of modifications to get this example working. SAM does not allow CLI command hooks. Amplify CLI allows that. In Amplify, you can create pre-push hooks or post-push hooks. 

This feature would have been useful since we are using Lambda Layers in this example in order to have minimal amount of package.jsons and also less bloating in each of the lambdas. In a typical Postgres application, multiple Lambdas will use the Knex package to interact with the database. So it makes sense to create a Lambda Layer for this.

## Summary by CodeRabbit

- **New Features**
  - Added a new `Users` class to handle user management with methods to add and retrieve users by ID.

- **Enhancements**
  - Updated build script and dependencies in `package.json` for better build and dependency management.
  - Introduced new configuration settings for deploying stacks in `samconfig.toml`.
  - Added new parameter and resource for common dependencies in `template.yaml`.

- **Documentation**
  - Updated `tsconfig.json` to correctly exclude the right node modules directory.

- **Bug Fixes**
  - Corrected `.gitignore` to properly ignore ZIP files and remove trailing characters.

### CODE REVIEW

File: src/users/users-module.ts [hunk 1]
```diff
@@ -0,0 +1,10 @@
+import PostgresDB from "../modules/psql";
+
+class Users extends PostgresDB {
+  constructor(connectionString: string) {
+    super(connectionString);
+  }
```
[the remark this answers is not in the document]
Comment (AkshathaLaxmi) [reply src/users/users-module.ts]: i pushed the fix in 904a81c

File: src/template.yaml [hunk 2]
```diff
@@ -69,6 +83,7 @@ Resources:
         Sourcemap: true
         EntryPoints: 
         - index.ts
+        External: !Split [" ", !Ref ExternalDependencies]
```
Comment (vishwasnavadak) [inline src/template.yaml:86]: This is a nice way of handling multiple values to be passed 👀

**Comment by AkshathaLaxmi** [comment]
@coderabbitai review

File: src/users/users-module.ts [hunk 3]
```diff
@@ -0,0 +1,39 @@
...
+class Users extends PostgresDB {
+    constructor(connectionString: string) {
+        super(connectionString);
+    }
+
+    
+    async addUser(username: string, email: string): Promise<void> {
+      // USER_TABLE environment variable contains the name of the table meant for storing user information
+      const id = randomUUID();
+      const userTable = process.env.USER_TABLE;
```
Comment (hnprashanth) [inline src/users/users-module.ts]: Would it be better to have userTable initialized when constructing the user class above?

File: src/users/users-module.ts [hunk 4]
```diff
@@ -0,0 +1,39 @@
...
+      }
+      return await this.getItem(userTable, id) as {id: number, username: string, email: string} | null;
+    }
+
+    // Add additional methods or override existing ones as needed
+
+}
+
+// PSQL_CONNECTION_STRING environment variable contains the connection string for the PostgreSQL database
+const connectionString = process.env.PSQL_CONNECTION_STRING;
```
Comment (hnprashanth) [inline src/users/users-module.ts:31]: this one as well...

**Review by hnprashanth — COMMENTED** [review]
Have a look at comments

File: src/users/users-module.ts (hunk 4 above)
Comment (vishwasnavadak) [reply src/users/users-module.ts:31]: If connectionString is being passed to constructor, why this is necessary?

Comment (AkshathaLaxmi) [reply src/users/users-module.ts:31]: Yes, is it better not to send any args to the constructor, and instead initialize a class variable with the connection string? I don't see any issues with either approach but what do you all feel?

File: src/users/users-module.ts (hunk 3 above)
Comment (AkshathaLaxmi) [reply src/users/users-module.ts]: done

File: src/users/users-module.ts (hunk 4 above)
Comment (vishwasnavadak) [reply src/users/users-module.ts:31]: My inclination is towards making these part of constructor params and not have anything on this. you can check if that is the pattern followed in any example

**Review by hnprashanth — APPROVED** [review]

**Pull request merged**